presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java

```diff
@@ -241,6 +252,37 @@ private DistributedQueryRunner(
         }
     }
 
+    public void waitForClusterToGetReady()
+            throws InterruptedException
+    {
+        HttpClient client = new JettyHttpClient();
+        for (int i = 0; i < coordinators.size(); i++) {
+            NodeState state = NodeState.INACTIVE;
+            while (state != NodeState.ACTIVE) {
+                MILLISECONDS.sleep(10);
+                state = getCoordinatorInfoState(client, i);
+            }
+        }
+
+        int availableCoordinaors = 0;
+        while (availableCoordinaors != coordinators.size()) {
+            MILLISECONDS.sleep(10);
+            availableCoordinaors = getResourceManager().get().getNodeManager().getCoordinators().size();
+        }
+    }
+
+    private NodeState getCoordinatorInfoState(HttpClient client, int coordinator)
+    {
+        URI uri = URI.create(getCoordinator(coordinator).getBaseUrl().toString() + "/v1/info/state");
+        Request request = prepareGet()
+                .setHeader(PRESTO_USER, "user")
```

Comment:
Instead of hardcoding the user here, can we be using the same user who is part of the session in the Runner?